src/components/accounts/create/ActivateAccount.js

```diff
@@ -142,45 +148,43 @@ class ActivateAccount extends Component {
         await dispatch(getAccountHelperWalletState(accountId));
     }
 
-    componentDidUpdate = (prevProps) => {
-        const { balance, minBalanceToUnlock, needsDeposit } = this.props;
-
-        if (minBalanceToUnlock && balance?.available !== prevProps.balance?.available) {
-            if (new BN(balance.available).gte(new BN(minBalanceToUnlock))) {
-                this.setState({ accountFunded: true });
-                window.scrollTo(0, 0);
-
-                if (needsDeposit) {
-                    this.handleClearAccountNeedsDeposit();
-                }
-            }
-        }
-    }
-
     componentWillUnmount = () => {
         this.stopPollingAccountBalance();
     }
 
-    handleClaimAccount = () => {
+    handleClaimAccount = async () => {
         const { dispatch, accountId, needsDeposit } = this.props;
 
+        this.setState({ activatingAccount: true });
+
         if (needsDeposit) {
-            this.handleClearAccountNeedsDeposit();
+            try {
+                await this.handleClearAccountNeedsDeposit();
+            } catch(e) {
+                this.setState({ activatingAccount: false });
+                throw e;
```

Comment:
Not a blocker, but I think this will bubble up and just be printed to the console as an un-handled exception.  Do we want to update this branch to display the error as a notification rather than just letting it bubble up to the browser (and become a Sentry report...)